numpy/lib/function_base.py

```diff
@@ -232,7 +232,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
                 tmp_a = tmp_a[keep]
                 if tmp_w is not None:
                     tmp_w = tmp_w[keep]
-            tmp_a = np.array(tmp_a, dtype=float, copy=False)
+            tmp_a = np.array(tmp_a, dtype=float, copy=True)
```

Comment:
No particular opinion on float versus np.float64. They do the same thing,
and I can't imagine this ever changing.
On Jul 27, 2015 14:12, "Jaime" <notifications@github.com> wrote:

> In numpy/lib/function_base.py
> <https://github.com/numpy/numpy/pull/6126#discussion_r35587778>:
>
> > @@ -232,7 +232,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
> >                  tmp_a = tmp_a[keep]
> >                  if tmp_w is not None:
> >                      tmp_w = tmp_w[keep]
> > -            tmp_a = np.array(tmp_a, dtype=float, copy=False)
> > +            tmp_a = np.array(tmp_a, dtype=float, copy=True)
>
> I think the intention of that line is better conveyed using the astype
> method, which defaults to copy=True, probably not worth setting it
> explicitly. Also, I have a feeling it is better to use np.double or
> np.float64 rather than float, but perhaps @njsmith
> <https://github.com/njsmith> has some thoughts on the proper thing to use
> from the recent discussion on deprecating np.float and friends.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/6126/files#r35587778>.
>